src/Send/Context/SendContext.tsx

```diff
@@ -39,114 +76,77 @@ export const SendProvider: React.FC<SendProvider> = ({children, balances, wallet
 
 const initialState: SendState = {
   sendAll: false,
+  primaryTokenId: '',
   selectedTokenId: '',
   receiver: '',
   amount: '',
 }
 
 const sendReducer = (state, action) => {
   switch (action.type) {
-    case SendActionTypes.SET_SELECTED_TOKEN_ID:
+    case 'changeSelectedTokenId':
       return {
         ...state,
         selectedTokenId: action.payload.selectedTokenId,
       }
-    case SendActionTypes.SET_RECEIVER:
+    case 'changeReceiver':
       return {
         ...state,
         receiver: action.payload.receiver,
       }
-    case SendActionTypes.SET_SEND_ALL:
+    case 'changeAmount':
       return {
         ...state,
-        sendAll: action.payload.sendAll,
+        amount: action.payload.amount,
       }
-    case SendActionTypes.SET_AMOUNT:
+    case 'toggleSendAll':
```

Comment:
avoid toggle

Reply:
suggestion?

Reply:
id consider putting this into `amountChanged`

Reply:
if `null` => sendAll ?

Reply:
Done